### PR TITLE
[Enhancement][be] add DEBUG_MODE in be_entrypoint.sh

### PR DESCRIPTION
**Pull request by mingmxu**
Add a new scenario(`DEBUG_MODE:true`) in `be_entrypoint.sh` to repeat launching BE process. Together with https://github.com/StarRocks/starrocks/pull/48866, they're tested in our SR clusters.

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docker/dockerfiles/be/be_entrypoint.sh [hunk 1]
```diff
@@ -132,10 +132,21 @@ while true; do
...
-  # If coredump is enabled, only catch SIGABRT (6), SIGSEGV(11) and repeat launching be process,
-  # should still exit for rest error codes.
-  if [[ "$COREDUMP_ENABLED" != "true" || ($ret -ne 134 && $ret -ne 139) ]]; then
-    exit $ret
+  # Repeat launching BE process in the two scenarios:
+  #  a. DEBUG_MODE is true;
+  #  b. coredump is enabled, and the error code is SIGABRT(6) or SIGSEGV(11);
+  # otherwise BE process should still exit.
+  should_exit=true
+  if [[ "$DEBUG_MODE" == "true" ]]; then
```
Comment (kevincai) [inline docker/dockerfiles/be/be_entrypoint.sh:140]: we may consider list all the logic control env variables into a central place and have comments explain each of them.

This scatter added variables makes the script harder and harder to understand and to maintain.

Comment (mingmxu) [reply docker/dockerfiles/be/be_entrypoint.sh:140]: agree, any place to keep the information now?

Comment (kevincai) [reply docker/dockerfiles/be/be_entrypoint.sh:140]: can start with summary in comments at the top of the script.

Comment (mingmxu) [reply docker/dockerfiles/be/be_entrypoint.sh:145]: added comments for the ENV variables.

**Review by kevincai — APPROVED** [review]

**Review by alvin-phoenix-ai — APPROVED** [review]

**Pull request merged**